File: expense-api/src/main/java/com/expense/expensemanagement/model/CurrencyType.java
package com.expense.expensemanagement.model;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum CurrencyType {

    INR("inr"), USD("usd"), EUR("eur"), PLN("pln");
    private String symbol;

}

File: expense-ui/src/modal/CurrencyType.ts
export enum CurrencyType{
    INR='INR',
    EUR= 'EUR',
    USD = 'USD',
    PLN = 'PLN',
}

export function getSymbol(currencyType?: CurrencyType){
    switch(currencyType) {
        case CurrencyType.INR:
            return "₹";

        case CurrencyType.USD:
            return "$";

        case CurrencyType.EUR:
            return 'Є';

        case CurrencyType.PLN:
            return 'zł';

        default:
            return currencyType;
    }
}
